### DISCUSSION TITLE
CDEvents Receiver and Provider support

### DISCUSSION THREAD

**Discussion by adamkenihan** [fluxcd/flux2#4339] [Proposals]
I am currently working on a PoC for [cdevents](https://cdevents.dev/) integration with flux. The receiver will  take and verify a CDEvent such as a change merged event, and the provider would send a CDEvent such as an environment modified when alerted.

**Comment by stefanprodan** [comment]
Hey @adamkenihan it would great if you could post here the spec on how the Flux events would be converted to cdevents. I guess events sent to notification-controller by kustomize-controller and helm-controller will get converted to [Continuous Deployment Events](https://github.com/cdevents/spec/blob/v0.3.0/continuous-deployment.md#continuous-deployment-events) then sent to external alert providers. Looking forward to your proposal on how to map the Flux event fields to CD event. Thanks!

**Reply by makkes** [reply]
Potentially related: https://github.com/fluxcd/flux2/pull/2639#issuecomment-1105294387

**Reply by adamkenihan** [reply]
Hi Stefan. We are not converting Flux events to cdevents, rather implementing a new receiver to consume a CDEvent (in a CloudEvent format), and this receiver will trigger reconciliation upon receiving a specific CDEvents type (ex: dev.cdevents.change.merged). 
And the provider will send specific CDEvents when alerted (using the go sdk). We are unsure which CDEvent we will send and are open to suggestions.

**Comment by adamkenihan** [comment]
I have a simple PoC made for this receiver and provider now if it would be possible to show it at the flux community dev meetings?
